### PR TITLE
MueLu add greedy dirichlet option for kokkos CoalesceDropFactory

### PR DESCRIPTION
**Pull request by pohm01**
@trilinos/muelu

## Motivation
This addresses #9957 
It also makes Dirichlet boundary detection consistent with the scalar case as well as the non-kokkos version, through the use of `Utilities_kokkos::DetectDirichletRows` to detect Dirichlet nodes.

## Testing
This adds a unit test for greedy dirichlet.

### CODE REVIEW

File: packages/muelu/src/Graph/MatrixTransformation/MueLu_CoalesceDropFactory_kokkos_def.hpp [hunk 1]
```diff
@@ -312,74 +312,86 @@ namespace MueLu {
 
     // build the dof-based column map containing the local dof ids belonging to blkSize rows in matrix
     // the DofIds may not be sorted.
```
Comment (jhux2) [inline packages/muelu/src/Graph/MatrixTransformation/MueLu_CoalesceDropFactory_kokkos_def.hpp:314]: I guess this comment is no longer valid?

**Review by jhux2 — APPROVED** [review]
lgtm

File: packages/muelu/src/Graph/MatrixTransformation/MueLu_CoalesceDropFactory_kokkos_def.hpp (hunk 1 above)
Comment (pohm01) [reply packages/muelu/src/Graph/MatrixTransformation/MueLu_CoalesceDropFactory_kokkos_def.hpp]: Yep, forgot to update the comments. Have updated now

**Review by jhux2 — APPROVED** [review]
lgtm

**Pull request merged**